drivers/console/ipm_console.c

```diff
@@ -11,12 +11,62 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(ipm_console, CONFIG_IPM_LOG_LEVEL);
 
-static struct device *ipm_console_device;
+#if defined(CONFIG_IPM_CONSOLE_IMMEDIATE)
+struct device *ipm_dev;
+#else
+/*
+ * Stack for the message queue handling thread
+ */
+static K_THREAD_STACK_DEFINE(thread_stack, 1024);
+static struct k_thread thread_data;
+
+K_MSGQ_DEFINE(msg_queue, sizeof(char), CONFIG_IPM_CONSOLE_QUEUE_SIZE, 4);
+
+static void msgq_thread(void *arg1, void *arg2, void *arg3)
+{
+	struct device *ipm_dev = arg1;
+
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	LOG_DBG("started thread");
+
+	while (true) {
+		int character = 0;
+		int ret;
+
+		/* Get character from message queue */
+		k_msgq_get(&msg_queue, &character, K_FOREVER);
+
+		/* Blocking write over IPM */
+		ret = ipm_send(ipm_dev, 1, character, NULL, 0);
+		if (ret) {
+			LOG_ERR("Error sending character %c over IPM, ret %d",
+				character, ret);
+		}
+
+		k_yield();
```

Comment:
Why?  This loop is going to jump straight into k_msgq_get() and block.  Please remove this yield, or if it's really needed let's figure out the root cause of the hack (likely a thread priority messup) and fix that instead.